In [12]:
import math

def dist(register):
    path=f'/scratch/project_462000353/amanda/register-training/register-model-training/sampling/distribution_results/eng_Latn_{register}.tsv'
    bins = []
    max_value = 0

    with open(path, 'r') as f:
        for line in f:
            num = int(line.split("\t")[1])
            if num>max_value: max_value=num
            if num==0: num+=1
            bin_to_put = int(math.log10(num))
            if len(bins) < bin_to_put + 1:
                for i in range(len(bins), bin_to_put+1):
                    bins.append(0)
            bins[bin_to_put] += 1

    values = [10**(i+1) for i in range(len(bins))]
    v_0 = 0
    print(f"Register: {register}")
    for v,b in zip(values, bins):
        print(f'{v_0}-{v}: \t {b}')
        v_0 = v
    print(f'maximum length: {max_value}')


for register in ["dtp", "HI", "HI-IN","IN","IP","LY", "NA", "ne", "no-label", "OP", "SP"]:
    dist(register)
    print("------------------------------")

Register: dtp
0-10: 	 1005
10-100: 	 57633225
100-1000: 	 143670702
1000-10000: 	 16593220
10000-100000: 	 504447
100000-1000000: 	 7663
maximum length: 546144
------------------------------
Register: HI
0-10: 	 252
10-100: 	 15457154
100-1000: 	 93042494
1000-10000: 	 11668065
10000-100000: 	 114525
100000-1000000: 	 197
maximum length: 301323
------------------------------
Register: HI-IN
0-10: 	 294
10-100: 	 13570831
100-1000: 	 52113325
1000-10000: 	 11665789
10000-100000: 	 136530
100000-1000000: 	 549
maximum length: 425541
------------------------------
Register: IN
0-10: 	 9824
10-100: 	 29366956
100-1000: 	 77725706
1000-10000: 	 19337950
10000-100000: 	 1223561
100000-1000000: 	 7702
maximum length: 602104
------------------------------
Register: IP
0-10: 	 10216
10-100: 	 65423765
100-1000: 	 112588461
1000-10000: 	 14301012
10000-100000: 	 84456
100000-1000000: 	 731
maximum length: 374390
------------------------------
Register: LY
0-10: 	 215
10-100: 	 3544589
100-1000: 

FileNotFoundError: [Errno 2] No such file or directory: '/scratch/project_462000353/amanda/register-training/register-model-training/sampling/distribution_results/eng_Latn_OP.tsv'

Register: dtp
0-10: 	 1005
10-100: 	 57633225
100-1000: 	 143670702
1000-10000: 	 16593220
10000-100000: 	 504447
100000-1000000: 	 7663
maximum length: 546144
------------------------------
Register: HI
0-10: 	 252
10-100: 	 15457154
100-1000: 	 93042494
1000-10000: 	 11668065
10000-100000: 	 114525
100000-1000000: 	 197
maximum length: 301323
------------------------------
Register: HI-IN
0-10: 	 294
10-100: 	 13570831
100-1000: 	 52113325
1000-10000: 	 11665789
10000-100000: 	 136530
100000-1000000: 	 549
maximum length: 425541
------------------------------
Register: IN
0-10: 	 9824
10-100: 	 29366956
100-1000: 	 77725706
1000-10000: 	 19337950
10000-100000: 	 1223561
100000-1000000: 	 7702
maximum length: 602104
------------------------------
Register: IP
0-10: 	 10216
10-100: 	 65423765
100-1000: 	 112588461
1000-10000: 	 14301012
10000-100000: 	 84456
100000-1000000: 	 731
maximum length: 374390
------------------------------
Register: LY
0-10: 	 215
10-100: 	 3544589
100-1000: 	 26388849
1000-10000: 	 1575848
10000-100000: 	 52963
100000-1000000: 	 290
maximum length: 194578
------------------------------
Register: NA
0-10: 	 17445
10-100: 	 34017323
100-1000: 	 120398659
1000-10000: 	 23117314
10000-100000: 	 658430
100000-1000000: 	 2277
maximum length: 610148
------------------------------
Register: ne
0-10: 	 2559
10-100: 	 24061593
100-1000: 	 120777126
1000-10000: 	 11870782
10000-100000: 	 202708
100000-1000000: 	 600
maximum length: 558550
------------------------------
Register: no-label
0-10: 	 108511
10-100: 	 33613921
100-1000: 	 109393648
1000-10000: 	 22377860
10000-100000: 	 331155
100000-1000000: 	 2551
maximum length: 970711
------------------------------


Problems were encountered in IN, no-label, NA and dtp. These visibly have the longest documents.
Limit seems to be around 550k. Let's try filtering IN and no-label based on that.